src/mongocxx/hint.hpp

```diff
@@ -70,8 +60,7 @@ class MONGOCXX_API hint {
     MONGOCXX_INLINE operator bsoncxx::document::value() const;
```

Comment:
should `friend bool operator==(const hint& index_hint, bsoncxx::document::view index);` also take a `view_or_value`?